Project/Assets/Scripts/Player/PlayerAttack.cs

```diff
@@ -69,30 +72,38 @@ public class PlayerAttack : MonoBehaviour {
             meleeAttackTrigger.enabled = false;
         }
     }
+    #endregion
 
-    private void CastAbility(Abilities ability)
+    //Casts players ability
+    private void CastAbility(AbilityTest ability)
     {
-        if (ability.isOffCooldown)
+        //Check to see ability is off cooldown
+        if (ability.offCooldown && (player.playerHUD.energySlider.value - ability.energy) > 0)
         {
-            anim.Play(ability.InputTag, 0);
+            //play the correct animation
+            anim.Play(ability.animationTag, 0);
+            //set the correct trigger
             meleeAttackTrigger.enabled = true;
-            meleeAttackTrigger.SendMessage("updateDamage", ability.Damage);
```

Comment:
Might be nice to see this message as a STATIC type somewhere (would define in c, but not c#) however, I'm flexable on this issue.

Pros: Reusable. If name changes, don't have to look everywhere for it.
Cons: String is a little obscured. Little more overhead.